ersilia/utils/identifiers/compound.py

```diff
@@ -122,25 +124,41 @@ def chemical_identifier_resolver(identifier):
 
     def encode(self, smiles):
         """Get InChIKey of compound based on SMILES string"""
+        if smiles is None or not smiles.strip():
+            return self.UNPROCESSABLE_INPUT
+        
         if self.Chem is None:
-            inchikey = self._pubchem_smiles_to_inchikey(smiles)
-            if inchikey is None:
-                inchikey = self._nci_smiles_to_inchikey(smiles)
+            inchikey = self._pubchem_smiles_to_inchikey(smiles) or self._nci_smiles_to_inchikey(smiles)
         else:
             try:
                 mol = self.Chem.MolFromSmiles(smiles)
                 if mol is None:
-                    raise Exception(
-                        "The SMILES string: %s is not valid or could not be converted to an InChIKey"
-                        % smiles
-                    )
+                    return self.UNPROCESSABLE_INPUT
```

Comment:
@musasizivictoria , You should ensure before returning UNPROCESSABLE_INPUT, the SMILES string is also processed by the other services and not just RDKit.